service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -716,6 +717,21 @@ public String cloneWorkspace(
         cloningInstructions,
         userRequest);
 
+    // Create the cloud context synchronously so that we can display a shell workspace.
```

Comment:
The `CreateWorkspaceV2Flight` creates the cloud context as part of the flight. Would it make sense to move the `createWorkspaceForClone` call above over to the v2 flight instead?

Reply:
Possibly… let me go through them both carefully and make sure it wouldn't change any behavior. I would need to determine the spend profile to use before calling the create workspace flight.